### PR TITLE
Remove bazel-based ./hack scripts

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This removes the not working scripts and changes the Makefile to not
reference them any more. `make verify` now tests via golangci-lint.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Removed scripts using bazel from ./hack directory
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Requires https://github.com/kubernetes/release/pull/1555

**Review by hasheddan — APPROVED** [review]
/lgtm

(After #1555 merges)

**Comment by saschagrunert** [comment]
`panic: test timed out after 10m0s`

File: hack/verify-golangci-lint.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.23.8
+VERSION=v1.31.0
```
Comment (justaugustus) [inline hack/verify-golangci-lint.sh:21]: This should be captured in `dependencies.yaml`.

**Review by justaugustus — CHANGES_REQUESTED** [review]
Awesome! Just one nit.

File: hack/verify-golangci-lint.sh (hunk 1 above)
Comment (saschagrunert) [reply hack/verify-golangci-lint.sh:21]: Added :+1:

**Pull request merged**